source/ota_interface.c

```diff
@@ -64,65 +76,87 @@ void setControlInterface( OtaControlInterface_t * pControlInterface )
 OtaErr_t setDataInterface( OtaDataInterface_t * pDataInterface,
                            const uint8_t * pProtocol )
 {
-    OtaErr_t err = OtaErrInvalidDataProtocol;
-    uint32_t i;
-
-    /*
-     * Primary data protocol will be the protocol used for file download if more
-     * than one protocol is selected while creating OTA job.
-     */
-    #if ( configOTA_PRIMARY_DATA_PROTOCOL == OTA_DATA_OVER_MQTT )
-        const char * pProtocolPriority[ OTA_DATA_NUM_PROTOCOLS ] =
+    OtaErr_t err = OtaErrNone;
+    bool httpInJobDoc;
+    bool mqttInJobDoc;
+
+    /* The explicit type casts create additional branches tracked by code
+     * coverage tools that are unreachable. These macros prevent the tools
+     * from tracking branch coverage for these lines. */
+    /* LCOV_EXCL_BR_START */
+    httpInJobDoc = ( strstr( ( const char * ) pProtocol, "HTTP" ) != NULL ) ? true : false;
```

Comment:
Why are we using `strstr()` rather than `strcmp()`?  This will set `httpInJobDoc == true` for `pProtocol="FOOHTTP123";` which doesn't seem correct.  I'd suggest something like:

```c
static const char * pOtaTransferTypes =
{
    /* Do not change the order of these */
    "MQTT",
    "HTTP",
    "MQTTHTTP",
    "HTTPMQTT"
};
static size_t numTransferTypes = sizeof( pOtaTransferTypes ) / sizeof( const char * );
int i;
for( i = 0; i < numTransferTypes; i++ )
{
    if( strcmp( pOtaTransferTypes, ( const char * ) pProtocol ) == 0 )
    {
        break;
    }
}

switch( i )
{
    case 0:   /* MQTT only, you could also use a #define/enum for this value */
         break;
    case 1:
        .
        .
        .
    default:  /* error handling goes here for everything > 3 */
        break;
}
```

nit: also combine assignment with declaration of `httpInJobDoc` & `mqttInJobDoc` since they're immediately assigned.

Reply:
This degree of leniency was how the original implementation worked. I didn't want to update the functionality, just the code coverage/implementation. I'm not as concerned about getting strings like "MQTTasdf" due to 1) the data coming from a fixed service that sends consist and 2) the error side effect of parsing that string would still be to set the data interface to MQTT which seems okay to me.

But it makes sense to have consistent and well defined behavior so I'm okay to make this change if you're okay with the behavior of setDataInterface being different from what it was before.